src/control/actor/host/req_metadata.go

```
@@ -0,0 +1,159 @@
+	var resCode reqresp.ResponseCode // error by default
+	err = methods.StatusRPCv1.RunRequest(ctx, h.NewStream, peerID, comp,
+		reqresp.RequestSSZInput{Obj: &data}, 1,
+		func() error {
+			return nil
+		},
+		func(chunk reqresp.ChunkedResponseHandler) error {
+			resCode = chunk.ResultCode()
+			switch resCode {
+			case reqresp.ServerErrCode, reqresp.InvalidReqCode:
+				msg, err := chunk.ReadErrMsg()
+				if err != nil {
+					return err
+				}
+				return errors.Errorf("error requesting BeaconState RPC: %s", msg)
+			case reqresp.SuccessCode:
+				var stat beacon.Status
+				if err := chunk.ReadObj(&stat); err != nil {
+					return err
+				}
+				data = stat
+			default:
+				return errors.New("unexpected result code")
+			}
+			return nil
+		})
+	return
+}
+
+func ReqBeaconMetadata(ctx context.Context, h host.Host, peerID peer.ID) (data beacon.MetaData, err error) {
+	// Generate the compression
+	comp := reqresp.SnappyCompression{}
+	// Generate the Server Error Code
+	var resCode reqresp.ResponseCode // error by default
+	err = methods.MetaDataRPCv1.RunRequest(ctx, h.NewStream, peerID, comp, reqresp.RequestSSZInput{Obj: &data}, 1,
+		func() error {
+			return nil
+		},
+		func(chunk reqresp.ChunkedResponseHandler) error {
+			resCode = chunk.ResultCode()
+			switch resCode {
+			case reqresp.ServerErrCode, reqresp.InvalidReqCode:
+				msg, err := chunk.ReadErrMsg()
+				if err != nil {
+					return errors.Errorf("error requesting BeaconMetadata RPC: %s", msg)
+				}
+			case reqresp.SuccessCode:
+				var meta beacon.MetaData
+				if err := chunk.ReadObj(&meta); err != nil {
+					return err
+				}
+				data = meta
+			default:
+				return errors.New("unexpected result code")
+			}
+			return nil
+		})
+	return
+}
+
+type HostWithIDService interface {
+	IDService() *identify.IDService
+}
+
+// ReqHostInfo request the host infomartion regarding a given peer, from the libp2p perspective
+// return empty struct and error if failure
```

perhaps its better to actually return an error if there is a failure.

---

I will add an error flag in case the libp2p identify request fails so we can also log it.